craft_application/models/project.py

```diff
@@ -131,13 +130,14 @@ def _vectorise_architectures(cls, values: str | list[str]) -> list[str]:
     @pydantic.field_validator("build_on", "build_for", mode="after")
     @classmethod
     def _validate_architectures(cls, values: list[str]) -> list[str]:
-        """Validate the architecture entries."""
-        for architecture in values:
-            if architecture != "all" and not is_valid_architecture(architecture):
-                raise errors.CraftValidationError(
-                    f"Invalid architecture: {architecture!r} "
-                    "must be a valid debian architecture."
-                )
+        """Validate the architecture entries.
+
+        Entries must be a valid debian architecture or 'all'. Architectures
+        may have an optional base prefix formatted as '[<base>:]<arch>'.
```

Comment:
Nitpick on naming: the architecture should be only the part after the colon, so you have the format `base:architecture` made of two separate entities. A more precise wording for this could be "Architectures may be preceded by an optional..." otherwise it would seem that the base is part of the architecture.

Reply:
Reworded